Sources/SpeziAccessGuard/AccessGuardViewModel.swift

```diff
@@ -101,7 +101,8 @@ final class AccessGuardViewModel {
             print("Error resetting access code: \(error)")
         }
     }
-    
+
+    @MainActor
```

Comment:
This already runs on the Main Thread. There is no need to do annotate this function.

Reply:
i'd argue that we should add the `@MainActor` isolation, but then at the same time also get rid of the `try await MainActor.run` inside the function (same for the other function below as well). that way the function's isolation would be transparent to the caller, instead of the current situation where it's de-facto MainActor-isolated, but the caller doesn't know this.

though i'd argue if we're already making basically every function in here `@MainActor`-isolated, we might as well just isolate the class as a whole to the MainActor (unless there's a reason why it isn't). that'd also allow us to make it Sendable (notwithstanding the question of whether that'd be something we'd actually want)

Reply:
As without the `@MainActor` isolation, we would run in a similar problem as discussed above. Therefore I'd also argue for isolating the entire `AccessGuardViewModel`. I added the changes, but left the `lock` method `async`, to not introduce any breaking API changes.